src/exceedances/job/views/exceedance.py

```diff
@@ -0,0 +1,387 @@
+                        'time': {'$lte': end, '$gte': start}}}
+
+    if pollutant == 'PM 10':
+        projection = {'$project': {'_id': 0,
+                                   'time': {'$dateToString': {'format': '%Y-%m-%dT%H:%M:%S%z', 'date': '$time', 'timezone': 'Africa/Kampala'}},
+                                   'pollutant_value': {'$round': ['$characteristics.pm10ConcMass.value', 2]}}}
+    elif pollutant == 'NO2':
+        projection = {'$project': {'_id': 0,
+                                   'time': {'$dateToString': {'format': '%Y-%m-%dT%H:%M:%S%z', 'date': '$time', 'timezone': 'Africa/Kampala'}},
+                                   'pollutant_value': {'$round': ['$characteristics.no2Conc.value', 2]}}}
+    else:
+        projection = {'$project': {'_id': 0,
+                                   'time': {'$dateToString': {'format': '%Y-%m-%dT%H:%M:%S%z', 'date': '$time', 'timezone': 'Africa/Kampala'}},
+                                   'pollutant_value': {'$round': ['$characteristics.pm2_5ConcMass.value', 2]}}}
+
+    if frequency == 'hourly':
+        # shall just query from the Events collection accordingly.
+        records = db.device_hourly_measurements.aggregate([query, projection])
+    elif frequency == 'monthly':
+        results = list(
+            db.device_daily_measurements.aggregate([query, projection]))
+        # because of the way data is stored, there will be no need to resample
+        records = resample_timeseries_data(results, 'M', 'time', 2)
+    else:
+        records = db.device_daily_measurements.aggregate([query, projection])
+
+    return list(records)
+
+
+def save_device_daily_exceedences(data):
+    """
+    # this ones moves to the exceedances model/entity
+    """
+    for i in data:
+        db.device_daily_exceedences.insert_one(i)
+
+
+def get_filtered_data(device_code, start_date=None, end_date=None, frequency='daily', pollutant='PM 2.5'):
+    """
+    returns the data of a certain device with specified parameters
+    this can also be part of the Events entity
+    """
+
+    if start_date == None:
+        start = convert_date.str_to_date_find('2019-06-01T00:00:00Z')
+    else:
+        start = convert_date.str_to_date(start_date)
+    if end_date == None:
+        end = datetime.now()
+    else:
+        end = convert_date.str_to_date(end_date)
+
+    query = {'deviceCode': device_code, 'time': {'$lte': end, '$gte': start}}
+
+    if pollutant == 'PM 10':
+        projection = {'_id': 0, 'time': 1,
+                      'characteristics.pm10ConcMass.value': 1}
+        if frequency == 'hourly':
+            records = list(
+                db.device_hourly_measurements.find(query, projection))
+        elif frequency == 'daily':
+            records = list(
+                db.device_daily_measurements.find(query, projection))
+        else:
+            records = list(db.device_raw_measurements.find(query, projection))
+        for i in range(len(records)):
```

Comment:
This is where the most important logic is for calculation, where the exceedances are calculated based on some predetermined thresholds, one for WHO and another one just AQI. @ja-odur